### PR TITLE
dnsdist: Implement 'reload()' to rotate Log(Response)Action's log file

### LINKED ISSUE #10502: dnsdist: Ponder adding the ability to rotate the LogAction file
- Program: dnsdist
 - Issue type: Feature request

### Short description

Some users are using `LogAction` for long-term logging of queries, keeping it enabled at all time instead of only for debugging purpose. For that use-case it would be nice to be able to ask for a rotation of the log file from time to time, via the console.

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Fixes #10502.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdist-lua-actions.cc [hunk 1]
```diff
@@ -806,46 +797,68 @@ class LogAction : public DNSAction, public boost::noncopyable
...
+    std::shared_ptr<FILE> fp = nullptr;
+
+    if (d_append) {
+      fp = std::shared_ptr<FILE>(fopen(d_fname.c_str(), "a+"), fclose);
+    }
+    else {
+      fp = std::shared_ptr<FILE>(fopen(d_fname.c_str(), "w"), fclose);
+    }
+
+    if (!fp) {
```
Comment (pieterlexis) [inline pdns/dnsdist-lua-actions.cc:818]: would it make sense to log that we were unable to re-open the file?

Comment (rgacogne) [reply pdns/dnsdist-lua-actions.cc:818]: Yes! I have just done that :+1:

**Comment by pandaedward** [comment]
Hi team, quick question, how do I call the reload method of the `LogAction` please? It is not very well documented.

At the moment I have this:
```
addAction(AllRule(), LogAction("/var/log/dnsdist.log", false, true, false))
```

How do I configure the conf file if I want `dnsdist.log` to be rotated please?

**Comment by rgacogne** [comment]
You are asking a question on a closed pull request, which is likely to be unseen. I would suggest posting on the mailing-list or on IRC instead.
The short version is that `LogAction("/var/log/dnsdist.log", false, true, false)` returns on object on which you can call the `reload()` method:
```
mylogaction = LogAction("/var/log/dnsdist.log", false, true, false)
[...]
mylogaction:reload()
```

This can be done via the console: https://dnsdist.org/guides/console.html or in `maintenance`, for example: https://dnsdist.org/reference/config.html#maintenance

**Pull request merged**